src/java.base/share/classes/java/security/Security.java

```diff
@@ -144,10 +144,13 @@ private static void initialize() {
         }
 /*[ENDIF] CRIU_SUPPORT */
 
-        // Load FIPS properties.
-        boolean fipsEnabled = FIPSConfigurator.configureFIPS(props);
-        if (sdebug != null) {
-            sdebug.println(fipsEnabled ? "FIPS mode enabled.": "FIPS mode disabled.");
+        // Load restricted security mode properties.
+        if (RestrictedSecurity.shouldEnable()) {
+            boolean restrictedSecurityEnabled = RestrictedSecurity.configure(props);
+            if (sdebug != null) {
+                sdebug.println(restrictedSecurityEnabled ? "Restricted security mode enabled."
+                        : "Restricted security mode disabled.");
+            }
```

Comment:
This seems like a step backward. In an earlier version of this change, the check that `RestrictedSecurity.configure()` was called at most once was internal to that method. I don't think that responsibility belongs here (at the call site). Please restore the behavior by updating `RestrictedSecurity.configure()` to detect a second call and fail if that occurs.

Reply:
@keithc-ca 

Ok, I will restore the previous version of this Security.java, to remove the RestrictedSecurity.shouldEnable() check.

And for detecting second call in "RestrictedSecurity.configure()". Right now, the current behaviour is, when go into the configure() method, if the securityEnabled is already true, that mean it is a second call, so it will throw the runtime exception saying "it can't be initialized twice". If the securityEnabled is false, that mean, either it is the first call, which is good. Or it is the second call but shouldEnableSecurity is false, and in this, the configure() method do nothing and just return false, means the restricted security is not support and no need to enable, or the user did not set semeru.restrictedsecurity or semeru.fips property.